Validator.py

```diff
@@ -0,0 +1,95 @@
+import glob
+import seaborn as sns
+from prompts import *
+
+class Validator:
+    def __init__(self, bad=None, confusion_matrix=None):
+        if bad is None:
+            bad = 0
+        self.bad = bad
+        if confusion_matrix is None:
+            confusion_matrix = [[0,0,0,0,0],
+                                [0,0,0,0,0],
+                                [0,0,0,0,0],
+                                [0,0,0,0,0],
+                                [0,0,0,0,0]] 
+        self.confusion_matrix = confusion_matrix
+    
+    def response_number(self, response):
+        if "pie chart" in response.lower():
+            return 0
+        elif "bar chart" in response.lower():
+            return 1
+        elif "line graph" in response.lower():
+            return 2
+        elif "list" in response.lower():
+            return 3
+        elif "trash" in response.lower():
+            return 4
+        else:
+            self.bad += 1
+            return 4
+        
+    def validation_helper(self, model, message, image, correct_response):
+        i = self.response_number(correct_response)
+        j = self.response_number(model.get_response(message, image))
+        self.confusion_matrix[i][j] += 1
+
+    def validation(self, model):
```

Comment:
Ajde probaj ovu implementaciju, mislim da je isto samo mnogo krace i vise u python duhu:
def validation(self, model):
    categories = ["pie", "bar", "line", "table", "trash"]
    labels = ["pie chart", "bar chart", "line graph", "list", "trash"]
    ctr = 1

    for i, category in enumerate(categories):
        print(category)
        for filename in glob.glob(f'/notebooks/dataset/{category}/*.jpg'):
            if ctr > 2 * (i + 1):
                break
            im = Image.open(filename)
            self.validation_helper(model, orchestrator_prompt, im, labels[i])
            print(ctr)
            ctr += 1

    return ctr

Takodje hajde da se ctr zove smisleno